tests/sympc/tensor/grad/grad_functions_test.py

```diff
@@ -641,3 +614,109 @@ def test_forward(get_clients) -> None:
     s_mpc.backward()
 
     assert np.allclose(x_mpc.grad.get(), x_secret.grad, rtol=1e-2)
+
+
+def test_grad_maxpool_2d_dilation_error(get_clients) -> None:
+    parties = get_clients(2)
+
+    secret = torch.Tensor(
+        [[[0.23, 0.32, 0.423], [0.2, -0.3, -0.53], [0.32, 0.42, -100]]]
+    )
+
+    x = secret.share(parties=parties)
+
+    with pytest.raises(ValueError):
+        GradMaxPool2D.forward({}, x, kernel_size=2, dilation=2)
+
+    with pytest.raises(ValueError):
+        GradMaxPool2D.forward({}, x, kernel_size=2, dilation=(1, 2))
+
+
+POSSIBLE_CONFIGS_MAXPOOL_2D = [
```

Comment:
Maybe: Add a test with rectangular kernel size? A rectangular kernel is used sometimes. Like for example, speech recognition. 
In case it's not supported for some reason we can throw an error. But, I noticed that the Conv2d was accepting tuple or int as kernel size.